app/src/main/java/org/simple/clinic/editpatient/PatientEditScreenController.kt

```diff
@@ -57,30 +57,10 @@ class PatientEditScreenController @Inject constructor(
     @Named("date_for_user_input") private val dateOfBirthFormatter: DateTimeFormatter
 ) : ObservableTransformer<UiEvent, UiChange> {
 
-  private val disposables = CompositeDisposable()
-
-  /**
-   * We do not want the UI stream to end if the count of subscribers change
-   * midway while the merge() inside apply is going through all Ui changes.
-   * As a solution, we're going to use autoConnect(), but that also means
-   * that this Transformer's stream have to be disposed manually by the screen.
-   */
-  fun disposeOnDetach(ui: Ui) {
-    RxView.detaches(ui)
-        .take(1)
-        .subscribe {
-          disposables.clear()
-        }
-  }
-
   override fun apply(events: Observable<UiEvent>): ObservableSource<UiChange> {
-    val replayedEvents = events.compose(ReportAnalyticsEvents())
-        .replay()
-        .autoConnect(1) { d -> disposables += d }
+    val replayUntilScreenIsDestroyed = ReplayUntilScreenIsDestroyed(ReportAnalyticsEvents(), mergeWithOngoingEntryPatientEntryChanges())
```

Comment:
It feels that replay-until-screen-destroyed is only being applied to the two arguments to `ReplayUntilScreenIsDestroyed` and not to `events` parameter. Do you think this will be more understandable?

```kotlin
ReplayUntilScreenIsDestroyed(events, ReportAnalyticsEvents(), mergeWithPatientSummaryChanges())
```

Not too convinced with this though.

Reply:
Yes, but passing `events` to it is tricky since it itself is composed on the stream outside and passing events to the constructor is pointless since it will be ignored.

Reply:
Discussed offline: Try to get a cleaner API like

```kotlin
val replayedEvents = ReplayUntilScreenIsDestroyed(events)
        .compose(ReportAnalyticsEvents())
        .compose(mergeWithPatientSummaryChanges())
        .replay()
```